drivers/sensor/stm32/temp_stm32f401x.c

```diff
@@ -15,22 +15,105 @@
  *          as 45 degrees C.
  */
 
-#include <device.h>
+#define SYS_LOG_DOMAIN "TEMPSTM32F401"
+#define SYS_LOG_LEVEL CONFIG_SYS_LOG_SENSOR_LEVEL
+
+#include <drivers/clock_control/stm32_clock_control.h>
+#include <errno.h>
+#include <logging/sys_log.h>
+#include <misc/__assert.h>
 #include <sensor.h>
 
-/* TODO */
+#include "temp_stm32.h"
+
+#define DEV_CFG(dev)							\
+	((const struct temp_stm32_config*)(dev)->config->config_info)
+#define DEV_DATA(dev) ((struct temp_stm32_data *)(dev)->driver_data)
+
+/*
+ * See STM32F401xD, STM32F401xE datasheet 6.3.21 and chip
+ * reference manual and ST RM0368 chapter 11.
+ */
+#define STM32F401_V25		0.76f
+#define STM32F401_AVG_SLOPE	2.5f
+
+#define ADC_RESOLUTION_12BIT 0x00
+#define ADC_PRESCALER_PCLK_DIV_8 0x03
+#define ADC_SAMPLE_480_CYCLES 0x07
+#define ADC_ONE_CONVERSION 0x00
+
 static int temp_stm32f401x_sample_fetch(struct device *dev,
 					enum sensor_channel chan)
 {
+	const struct temp_stm32_config *cfg = DEV_CFG(dev);
+	ADC_TypeDef *adc = cfg->adc;
+	uint32_t tmp;
+
+	/* This is a valid assumption on STM32F401x. */
+	__ASSERT(cfg->adc_channel == 18,
```

Comment:
Can we make a #define for 18 and use it here as well as below in the temp_stm32f401x_config structure for adc_channel?